src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -2087,6 +2109,14 @@ export class SavedObjectsRepository {
     );
 
     const indexFound = statusCode !== 404;
+
+    // check if we have the elasticsearch header when doc.found is not true (false, undefined or null) and if we do, ensure it is Elasticsearch
+    const esHeaderFound = isEsHeaderValid(headers);
+
+    if (!isFoundGetResponse(body) && !esHeaderFound) {
```

Comment:
Question: Might it lead to false positives if an intermediate proxy doesn't forward `x-elastic-product` header? In `v7.14`, we do allow Kibana starts even if `x-elastic-product` header is missing. Kibana won't start unless the header presents starting from `v7.15` https://github.com/elastic/kibana/issues/105557
Does it mean that we cannot backport this change to `v7.14.1`?

Reply:
> Might it lead to false positives if an intermediate proxy doesn't forward `x-elastic-product` header?

If the header is missing from the response then yes, we would get `503` errors rather than `404` (which is what we're getting now). We would, however, still be getting an error because ATM, the header check is only done if we are already getting a `404` response. 

If we did backport the work to 7.14.1, and that was the original idea, then there is a chance we'd get false positives but I don't know how likely that would be. We rely on headers for other Kibana functionality (e.g. accessing internal indices). In what cases wouldn't a proxy forward response headers on?